stdlib/asm/crypto/fri/frie2f4.masm

```diff
@@ -270,28 +338,152 @@ export.verify.4
+    push.0.0 
+    exec.constants::tmp2 exec.constants::get_remainder_poly_address
+    exec.constants::tmp1 mem_storew
+    movup.4
+
     while.true
         # load [e0, e1, p, poe] from memory i.e. next query data (7 cycles)
-        push.0.0.0.0
         movup.4
         mem_loadw
         # => [poe, p, e1, e0, layer_ptr, rem_ptr, g, ...]
 
         # we now have everything to verify query p
-        exec.verify_query
+        exec.verify_query_64
 
         # prepare for next iteration (18 cycles)
         # => [x, x, x, x, x, x, x, x, x, x, g, ...]
-        dropw drop drop drop
+        dropw drop
         loc_loadw.0   # load [query_ptr, layer_ptr, rem_ptr, g]
         add.4
-        loc_storew.0  # store [query_ptr + 4, layer_ptr, rem_ptr, g]
-        dup
-        dup.2
+        loc_storew.0  # store [query_ptr + 1, layer_ptr, rem_ptr, g]
+        swapw
+        # => [x, x, x, x, query_ptr + 1, layer_ptr, rem_ptr, g, ...]
+        dup.5
+        dup.5
         neq
-        #=> [?, query_ptr + 4, layer_ptr, rem_ptr, g, ...]
+        #=> [?, query_ptr + 1, layer_ptr, rem_ptr, g, ...]
     end
     #=> [X, ..]
 
-    dropw
+    dropw dropw
+end
+
+#! Verifies a FRI proof where the proof was generated over the quadratic extension of the base
+#! field and layer folding was performed using folding factor 4 when the degree of the remainder
+#! polynomial is less than 128.
+#!
+#! Input:  [query_ptr, layer_ptr, rem_ptr, g, ...]
+#! Output: [...]
+#!
+#! - query_ptr is a pointer to a list of tuples of the form (e0, e1, p, poe) where poe is equal
+#!   to g^p with g being the initial FRI domain generator. p is the query index at the first layer
+#!   and (e0, e1) is an extension field element corresponding to the value of the first layer at index p.
+#! - layer_ptr is a pointer to the first layer commitment denoted throughout the code by C.
+#!   layer_ptr + 1 points to the first [alpha0, alpha1, t_depth, d_size] where d_size is the size
+#!   of initial domain divided by 4, t_depth is the depth of the Merkle tree commitment to the
+#!   first layer and (alpha0, alpha1) is the first challenge used in folding the first layer.
+#!   Both t_depth and d_size are expected to be smaller than 2^32. Otherwise, the result of
+#!   this procedure is undefined.
+#! - rem_ptr is a pointer to the first tuple of two consecutive degree 2 extension field
+#!   elements making up the remainder polynomial. This procedure is specialized to the case when
+#!   the the degree of the latter is less than 128.
+#!
+#! The memory referenced above is used contiguously, as follows:
+#!
+#!   [query_ptr ... layer_ptr ... rem_ptr ...]
+#!
+#! This means for example that:
+#! 1. rem_ptr - 1 points to the last (alpha0, alpha1, t_depth, d_size) tuple.
+#! 2. layer_ptr - 1 points to the last (e0, e1, p, poe) tuple.
+#!
+#! Cycles: 24 + num_queries * (140 + num_layers * 83)
+proc.verify_128.4
```

Comment:
`verify_128` and `verify_64`, as well as `verify_query_64` and `verify_query_128` are copy/pasted versions of each other, except for the number times we call `mem_stream horner_eval_ext`. I would add this to their corresponding docstrings just to help the reader (e.g. it took me a bit to confirm that this was actually the case).

Reply:
Good point, if they were not performance critical it would've made more sense to have only one procedure that has an if-else statement branching based on the size of the remainder.
I have added comments to clarify this point in the docstrings